### ISSUE TITLE
Enhance input processing

### ISSUE THREAD

**Issue by datech** [datech/node-red-contrib-amazon-echo#60]
Enhance input processing to support 3 different options:

- **No processing on the input** 
Input payloads will be ignored

- **Process**
Input payloads will be processed, and Amazon Echo Hub will save the states, but without outputting the payload

- **Process and output**
Input payloads will be processed, and output will be passed to all available devices.

**Comment by datech** [comment]
Thanks, @Barabba11 

I've checked the @verjus implementation. Mine is similar, and I've just summarized it in a single dropdown instead of two checkboxes.

**Comment by verjus** [comment]
Hello Datech,

First, I am very impressed with your code. Thank you for making it publicly available.

Please take a look at the small modifications I made in this [fork](https://github.com/verjus/node-red-contrib-amazon-echo). One modification achieves a similar result but dynamically, instead of relying on a static setting, by relying on the content of the topic in the input message. If the topic is empty, nothing is passed through in output (only the state is updated). If the topic is set (e.g. topic=brightness), the corresponding state is updated and passed in output. 

Other potentially interesting modifications are discussed [here](https://github.com/datech/node-red-contrib-amazon-echo/issues/55).

**Comment by Barabba11** [comment]
Can somebody kindly tell me how is useful to process an input payload but don't output it? How can be useful to update something on HUB? (what it updates? It's a just a transparent process to node red devices.). 
may Echo ask something to HUB about a status? Thank you

**Comment by verjus** [comment]
Alexa queries the state of the (emulated) hub when certain conditions are met. The state of your bulbs is kept in memory by this software. That state will be queried by Alexa over your local network and will be reflected in the Alexa app. However, if you use another method to change the state of a physical bulb, the physical state of the bulb will change but this software will not know about it. For example, I use a motion sensor and Philips switches. When I change the state of a bulb using the switch or because of the motion sensor, the software will not know about that change since it took place over the Zigbee network. Using Zigbee2mqtt, I can update the state in the software memory by sending this info in the input node. Now when Alexa queries the state of a given bulb, it will get the correct state. Another use case is if you use a node-red based timer to turn on your lights automatically at a certain time. You would also need to update the state of the lights in the software.

On a side node, that state is kept in a node context. This means that it can also be changed without using node-red wires, by modifying the state directly with a simple modification. So, I've modified the software to keep this information in a flow context instead by changing this line:

`setDeviceAttributes(msg.payload.deviceid, msg.payload, hubNode.context());`
to
`setDeviceAttributes(msg.payload.deviceid, msg.payload, hubNode.context().flow);`

This way, my node red sensor code looks like this:

```
const BULB= 'Osram';
const THRESHOLD=  5;
const BRIGHTNESS= 5;

var outMsg = { 'payload': '',
               'topic'  : '' };
var sensor= JSON.parse(msg.payload);

**state= flow.get(BULB) || {};**

if (typeof state.on === 'undefined') {
  state.on= false
}

//node.warn('last='+ last);
//node.warn('current='+ current);

switch (true) {

  // Turn On:
  case ( (sensor.occupancy === true) && (sensor.illuminance <= THRESHOLD) && (state.on === false) ): 
        outMsg.topic= 'zigbee2mqtt/' + BULB +'/set';
        outMsg.payload= {'state':'ON', 'brightness':BRIGHTNESS, 'transition':2};
        node.send(outMsg);
        **state.on= true;
        flow.set(BULB, state);**
        break;

  // Turn Off:        
  case (sensor.occupancy === false) && (state.on === true):
        outMsg.topic= 'zigbee2mqtt/' + BULB +'/set';
        outMsg.payload= {'state':'OFF', 'transition':4};
        node.send(outMsg);
        **state.on= false;
        flow.set(BULB, state);**
        break;

}
```

Perhaps this simple modification can be incorporated in the main branch.

Hope this helps!

**Comment by datech** [comment]
Released in v0.1.9
